### DISCUSSION TITLE
How to compile both .js and .ts?

### DISCUSSION THREAD

**Discussion by marcenacp** [swc-project/swc#1877] [Q&A] [answered]
I want to compile React code. How should I configure my `.swcrc` file in order to compile both `.ts`/`.tsx` and `.js` files? Thanks.

**Comment by marcenacp** [comment, marked as answer]
`.swrc` accepts a list of configurations, so you can add several parsers.

Example:

```
[
  {
    "test": ".tsx?$",
    "jsc": {
      "parser": {
        "syntax": "typescript",
        "tsx": true,
        "decorators": false,
        "dynamicImport": true
      },
      "transform": {
        "react": {
          "refresh": true
        }
      }
    }
  },
  {
    "test": ".jsx?$",
    "jsc": {
      "parser": {
        "syntax": "ecmascript",
        "jsx": true,
        "decorators": false,
        "dynamicImport": true
      },
      "transform": {
        "react": {
          "refresh": true
        }
      }
    }
  }
]
```

**Reply by mrtnpro** [reply]
How would you add a schema in this case? Examples taken from https://swc.rs/docs/configuration/swcrc#compilation
```
 {
  "$schema": "https://json.schemastore.org/swcrc"
}
```

**Reply by sidhuko** [reply]
Would be good to know how this can be passed to swc-loader too. I know it _can_ be done in webpack but it has certain performance costs to use multiple loaders especially when using with `thread-loader` and following webpack performance guides.

**Reply by fenghourun** [reply]
@marcenacp any way  to incllude the `schema` in this setup?

**Reply by hugotox** [reply]
This is no longer working. Any idea how to add this config for "@swc/core": "^1.4.2"?

**Reply by kdy1** [reply]
@hugotox Can you file an issue?

**Comment by wegry** [comment]
If you're using typescript and `@swc/core@1.4.14`, writing a "test" with both js and ts seems to work. This has the added benefit of allowing the schema to still be set.

```json
{
	"$schema": "https://json.schemastore.org/swcrc",
	"test": ".(j|t)sx?$",
	"jsc": {
		"parser": {
			"syntax": "typescript",
			"tsx": true
		},
		"transform": {
			"react": {
				"runtime": "automatic"
			}
		}
	}
}
```
